server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java

```diff
@@ -101,4 +103,26 @@ public static void verifyNoMultipleWriters(List<String> mdFiles, Function<String
         });
     }
 
+    /**
+     * Converts an input hash which occupies 64 bits of space into Base64 (6 bits per character) String. This must not
+     * be changed as it is used for creating path for storing remote store data on the remote store.
+     */
+    static String longToBase64(long value) {
+        byte[] hashBytes = ByteBuffer.allocate(Long.BYTES).putLong(value).array();
+        return base64(hashBytes);
+    }
+
+    /**
+     * This converts the byte array to base 64 string. `/` is replaced with `_`, `+` is replaced with `-` and `=`
+     * which is padded at the last is also removed. These characters are either used as delimiter or special character
+     * requiring special handling in some vendors. The characters present in this base64 version are [A-Za-z0-9_-].
+     * This must not be changed as it is used for creating path for storing remote store data on the remote store.
+     *
+     * @param bytes byte array
+     * @return base 64 string.
+     */
+    private static String base64(byte[] bytes) {
+        String base64 = Base64.getEncoder().encodeToString(bytes);
+        return base64.substring(0, base64.length() - 1).replace('/', '_').replace('+', '-');
+    }
```

Comment:
Just found out that there is an URL safe encoder. I will replace this with Java provided option.

Reply:
I will push the same after 1 round of review.